src/reanimated2/animation/delay.ts

```diff
@@ -31,6 +31,13 @@ export const withDelay = function <T extends AnimationObject>(
   reduceMotion?: ReduceMotion
 ): Animation<DelayAnimation> {
   'worklet';
+
+  const initialAnimationStart =
+    typeof _nextAnimation === 'function'
+      ? _nextAnimation().startValue
+      : _nextAnimation.startValue;
+  let didResetAfterInterruption = false;
```

Comment:
Is it correct to initialize `didResetAfterInterruption` with `false`? This means we will reset the animation at the beginning so before it could have been interrupted

Reply:
I think I should rename it to something like `shouldSkipNextValueSet`. This flag helps us prevent setting current value to the previous one (which makes resetting impossible).